Source/Scene/TerrainFillMesh.js

```diff
@@ -0,0 +1,1273 @@
+                // }
+                destinationFill.changedThisFrame = destinationFill.changedThisFrame || destinationFill.northeastMesh !== sourceMesh;
+                destinationFill.northeastMesh = sourceMesh;
+                destinationFill.northeastTile = sourceTile;
+                return;
+            case TileEdge.SOUTHWEST:
+                // if (destinationFill.southwestTile !== sourceTile) {
+                //     const from = destinationFill.southwestTile || {};
+                //     console.log(`L${destinationTile.level}X${destinationTile.x}Y${destinationTile.y} changed because southwest tile changed from L${from.level}X${from.x}Y${from.y} to L${sourceTile.level}X${sourceTile.x}Y${sourceTile.y}`);
+                // }
+                destinationFill.changedThisFrame = destinationFill.changedThisFrame || destinationFill.southwestMesh !== sourceMesh;
+                destinationFill.southwestMesh = sourceMesh;
+                destinationFill.southwestTile = sourceTile;
+                return;
+            case TileEdge.SOUTHEAST:
+                // if (destinationFill.southeastTile !== sourceTile) {
+                //     const from = destinationFill.southeastTile || {};
+                //     console.log(`L${destinationTile.level}X${destinationTile.x}Y${destinationTile.y} changed because southeast tile changed from L${from.level}X${from.x}Y${from.y} to L${sourceTile.level}X${sourceTile.x}Y${sourceTile.y}`);
+                // }
+                destinationFill.changedThisFrame = destinationFill.changedThisFrame || destinationFill.southeastMesh !== sourceMesh;
+                destinationFill.southeastMesh = sourceMesh;
+                destinationFill.southeastTile = sourceTile;
+                return;
+        }
+
+        if (sourceTile.level <= destinationTile.level) {
+            // Source edge completely spans the destination edge.
+            // if (edgeTiles[0] !== sourceTile) {
+            //     const from = edgeTiles[0] || {};
+            //     console.log(`L${destinationTile.level}X${destinationTile.x}Y${destinationTile.y} changed because edge tile changed from L${from.level}X${from.x}Y${from.y} to L${sourceTile.level}X${sourceTile.x}Y${sourceTile.y}`);
+            // }
+            destinationFill.changedThisFrame = destinationFill.changedThisFrame || edgeMeshes[0] !== sourceMesh || edgeMeshes.length !== 1;
+            edgeMeshes[0] = sourceMesh;
+            edgeTiles[0] = sourceTile;
+            edgeMeshes.length = 1;
+            edgeTiles.length = 1;
+            return;
+        }
+
+        // Source edge is a subset of the destination edge.
+        // Figure out the range of meshes we're replacing.
+        var startIndex, endIndex, existingTile, existingRectangle;
+        var sourceRectangle = sourceTile.rectangle;
+
+        function firstIsLessThanSecond(a, b, epsilon) {
```

Comment:
Do these beyond in `CesiumMath` with friends?

https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Math.js#L569